lib/ledger_sync/adaptors/quickbooks_online/bill/operations/create.rb

```diff
@@ -0,0 +1,65 @@
+module LedgerSync
+  module Adaptors
+    module QuickBooksOnline
+      module Bill
+        module Operations
+          class Create < Operation::Create
+            class Contract < LedgerSync::Adaptors::Contract
+              params do
+                required(:ledger_id).value(:nil)
+                required(:vendor).hash(Types::Reference)
+                required(:account).hash(Types::Reference)
+                required(:currency).filled(:string)
+                required(:memo).filled(:string)
+                required(:transaction_date).filled(:date?)
+                required(:due_date).filled(:date?)
+                required(:line_items).array(Types::Reference)
+              end
+            end
+
+            private
+
+            def operate
+              response = adaptor.post(
+                resource: 'bill',
+                payload: local_resource_data
+              )
+
+              resource.ledger_id = response.dig('Id')
+              success(response: response)
+            end
+
+            def local_resource_data
+              {
+                'CurrencyRef' => {
+                  'value' => resource.currency,
+                },
+                'TxnDate' => resource.transaction_date.to_s, # Format: YYYY-MM-DD
+                'DueDate' => resource.due_date.to_s, # Format: YYYY-MM-DD
+                'PrivateNote' => resource.memo,
+                'VendorRef' => {
+                  'value' => resource.vendor.ledger_id,
+                },
+                'APAccountRef' => {
+                  'value' => resource.account.ledger_id
+                },
+                'Line' => resource.line_items.map do |line_item|
+                  {
+                    'DetailType' => 'AccountBasedExpenseLineDetail',
```

Comment:
Yeah, Bill is using ExpenseLineDetail. I thought about reusing our existing line item, but different ledger could be using different structure. So let it be separate `BillLineItem`

Reply:
Yeah I think this makes sense